lib/ansible/plugins/lookup/fileglob.py

```diff
@@ -26,12 +26,10 @@ class LookupModule(LookupBase):
 
     def run(self, terms, variables=None, **kwargs):
 
-        basedir = self.get_basedir(variables)
-
         ret = []
         for term in terms:
             term_file = os.path.basename(term)
-            dwimmed_path = self._loader.path_dwim_relative(basedir, 'files', os.path.dirname(term))
+            dwimmed_path = self.find_needle(variables, 'files', os.path.dirname(term))
```

Comment:
Could 'files' be replaced with a @property (self.file_search_path?).  For these cases it would return ['files']
or  below it would be ['templates', 'vars', 'files']